# it is a regression problem 

## we are doing the  ml on house pricing 

<table align="center" color="#216180" width=100%>
    <tr>
        </td>
            </td>
        </td>
        <td>
            <div align="center">
                <font color="#216180" size=16px>
                    <b> Regression modeling on House Price Data
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

# Problem Statement

This dataset is created for the prediction of House Prices from an Indian perspective Towns. The dataset contains various features that are important during the application for the selling the Houses. The predicted output obtained from the classification algorithm gives a fair idea about the chances of a Houses Sold.

In [1]:
#import numpy
import numpy as np#to find statestical summery 
#import pandas
import pandas as pd#formate the data stacture (convert any data stucture into pandas data frame)
#import matplotlib
import matplotlib.pyplot as plt
from matplotlib.colors import ListedColormap
#import seaborn
import seaborn as sn

### load the data set

### load the data and name it df_house_price

In [2]:
#load csv file
#stire the data in df_houseprice
df_houseprice = pd.read_csv(r"C:\Users\hp\Downloads\House_Price1.csv")
df_houseprice

price  resid_area  air_qual  room_num   age  distance  teachers  \
0     24.0       32.31     0.538     6.575  65.2      4.35      24.7   
1     21.6       37.07     0.469     6.421  78.9      4.99      22.2   
2     34.7       37.07     0.469     7.185  61.1      5.03      22.2   
3     33.4       32.18     0.458     6.998  45.8      6.21      21.3   
4     36.2       32.18     0.458     7.147  54.2      6.16      21.3   
..     ...         ...       ...       ...   ...       ...       ...   
501   22.4       41.93     0.573     6.593  69.1      2.64      19.0   
502   20.6       41.93     0.573     6.120  76.7      2.44      19.0   
503   23.9       41.93     0.573     6.976  91.0      2.34      19.0   
504   22.0       41.93     0.573     6.794  89.3      2.54      19.0   
505   19.0       41.93     0.573     6.030  80.8      2.72      19.0   

     poor_prop airport  n_hos_beds  n_hot_rooms       waterbody  rainfall  \
0         4.98     YES       5.480      11.1920           River        23   
1         9.14      NO       7.332      12.1728            Lake        42   
2         4.03      NO       7.394     101.1200            None        38   
3         2.94     YES       9.268      11.2672            Lake        45   
4         5.33      NO       8.824      11.2896            Lake        55   
..         ...     ...         ...          ...             ...       ...   
501       9.67      NO       9.348      12.1792  Lake and River        27   
502       9.08     YES       6.612      13.1648  Lake and River        20   
503       5.64      NO       5.478      12.1912            None        31   
504       6.48     YES       7.940      15.1760            None        47   
505       7.88     YES      10.280      10.1520            None        45   

    bus_ter     parks  Sold  
0       YES  0.049347     0  
1       YES  0.046146     1  
2       YES  0.045764     0  
3       YES  0.047151     0  
4       YES  0.039474     0  
..      ...       ...   ...  
501     YES  0.056006     1  
502     YES  0.059903     1  
503     YES  0.057572     1  
504     YES  0.060694     1  
505     YES  0.060336     1  

[506 rows x 16 columns]

In [3]:
#name of columns
df_houseprice.columns

Index(['price', 'resid_area', 'air_qual', 'room_num', 'age', 'distance',
       'teachers', 'poor_prop', 'airport', 'n_hos_beds', 'n_hot_rooms',
       'waterbody', 'rainfall', 'bus_ter', 'parks', 'Sold'],
      dtype='object')

**interpretation**

 - `price` :price of the house.
 - `resid_area` :residentially area where my house is located

In [4]:
# basic info of the data set
df_houseprice.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 16 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   price        506 non-null    float64
 1   resid_area   506 non-null    float64
 2   air_qual     506 non-null    float64
 3   room_num     506 non-null    float64
 4   age          506 non-null    float64
 5   distance     506 non-null    float64
 6   teachers     506 non-null    float64
 7   poor_prop    506 non-null    float64
 8   airport      506 non-null    object 
 9   n_hos_beds   498 non-null    float64
 10  n_hot_rooms  506 non-null    float64
 11  waterbody    506 non-null    object 
 12  rainfall     506 non-null    int64  
 13  bus_ter      506 non-null    object 
 14  parks        506 non-null    float64
 15  Sold         506 non-null    int64  
dtypes: float64(11), int64(2), object(3)
memory usage: 63.4+ KB


**interpretation**
 - we have 13 numerical columns
 - we have NULL value in data set
 - we have 16 column in data

In [5]:
#shape of the data
df_houseprice.shape

(506, 16)

**interpretation**
 - we have total `16` columns
 - we have total `506` objervations

In [6]:
# summaraze the badic stastic of data
df_houseprice.describe()

price  resid_area    air_qual    room_num         age    distance  \
count  506.000000  506.000000  506.000000  506.000000  506.000000  506.000000   
mean    22.528854   41.136779    0.554695    6.284634   68.574901    3.971996   
std      9.182176    6.860353    0.115878    0.702617   28.148861    2.108532   
min      5.000000   30.460000    0.385000    3.561000    2.900000    1.130000   
25%     17.025000   35.190000    0.449000    5.885500   45.025000    2.270000   
50%     21.200000   39.690000    0.538000    6.208500   77.500000    3.385000   
75%     25.000000   48.100000    0.624000    6.623500   94.075000    5.367500   
max     50.000000   57.740000    0.871000    8.780000  100.000000   12.320000   

         teachers   poor_prop  n_hos_beds  n_hot_rooms    rainfall  \
count  506.000000  506.000000  498.000000   506.000000  506.000000   
mean    21.544466   12.653063    7.899767    13.041605   39.181818   
std      2.164946    7.141062    1.476683     5.238957   12.513697   
min     18.000000    1.730000    5.268000    10.057600    3.000000   
25%     19.800000    6.950000    6.634500    11.189800   28.000000   
50%     20.950000   11.360000    7.999000    12.720000   39.000000   
75%     22.600000   16.955000    9.088000    14.170800   50.000000   
max     27.400000   37.970000   10.876000   101.120000   60.000000   

            parks        Sold  
count  506.000000  506.000000  
mean     0.054454    0.454545  
std      0.010632    0.498422  
min      0.033292    0.000000  
25%      0.046464    0.000000  
50%      0.053507    0.000000  
75%      0.061397    1.000000  
max      0.086711    1.000000

**interpretation**
 - we have a outliers in the data most of the outlier aer in age column

## data preprocessing


## a. Data Cleaning

In [7]:
#show me 1st FIVE observation
df_houseprice.head()

price  resid_area  air_qual  room_num   age  distance  teachers  poor_prop  \
0   24.0       32.31     0.538     6.575  65.2      4.35      24.7       4.98   
1   21.6       37.07     0.469     6.421  78.9      4.99      22.2       9.14   
2   34.7       37.07     0.469     7.185  61.1      5.03      22.2       4.03   
3   33.4       32.18     0.458     6.998  45.8      6.21      21.3       2.94   
4   36.2       32.18     0.458     7.147  54.2      6.16      21.3       5.33   

  airport  n_hos_beds  n_hot_rooms waterbody  rainfall bus_ter     parks  Sold  
0     YES       5.480      11.1920     River        23     YES  0.049347     0  
1      NO       7.332      12.1728      Lake        42     YES  0.046146     1  
2      NO       7.394     101.1200      None        38     YES  0.045764     0  
3     YES       9.268      11.2672      Lake        45     YES  0.047151     0  
4      NO       8.824      11.2896      Lake        55     YES  0.039474     0

In [8]:
#step2 alllocate new colum name's to old colums name
#make the Dictonary{"old column name":"new column name"}
D1 ={'price' : 'Price',
     'resid_area' : 'Residential_Area',
     'air_qual': 'Air_Qulity',
    'room_num': 'Room_Numbers',
    'age' : 'Age',
    'distance' : 'Distance',
    'teachers':'Teachers',
    'poor_prop': 'Poor_Population',
    'airport':'Airport',
    'n_hos_beds' : 'Number_Of_Hospital_Beds',
    'waterbody' : 'Waterbody',
    'rainfall':'Rainfall',  
   'bus_ter':'Bus_Terminal',
    'parks':'Parks',
    'sold':'Sold'}
D1

{'price': 'Price',
 'resid_area': 'Residential_Area',
 'air_qual': 'Air_Qulity',
 'room_num': 'Room_Numbers',
 'age': 'Age',
 'distance': 'Distance',
 'teachers': 'Teachers',
 'poor_prop': 'Poor_Population',
 'airport': 'Airport',
 'n_hos_beds': 'Number_Of_Hospital_Beds',
 'waterbody': 'Waterbody',
 'rainfall': 'Rainfall',
 'bus_ter': 'Bus_Terminal',
 'parks': 'Parks',
 'sold': 'Sold'}

In [9]:
df_houseprice = df_houseprice.rename(columns = D1)
df_houseprice

Price  Residential_Area  Air_Qulity  Room_Numbers   Age  Distance  \
0     24.0             32.31       0.538         6.575  65.2      4.35   
1     21.6             37.07       0.469         6.421  78.9      4.99   
2     34.7             37.07       0.469         7.185  61.1      5.03   
3     33.4             32.18       0.458         6.998  45.8      6.21   
4     36.2             32.18       0.458         7.147  54.2      6.16   
..     ...               ...         ...           ...   ...       ...   
501   22.4             41.93       0.573         6.593  69.1      2.64   
502   20.6             41.93       0.573         6.120  76.7      2.44   
503   23.9             41.93       0.573         6.976  91.0      2.34   
504   22.0             41.93       0.573         6.794  89.3      2.54   
505   19.0             41.93       0.573         6.030  80.8      2.72   

     Teachers  Poor_Population Airport  Number_Of_Hospital_Beds  n_hot_rooms  \
0        24.7             4.98     YES                    5.480      11.1920   
1        22.2             9.14      NO                    7.332      12.1728   
2        22.2             4.03      NO                    7.394     101.1200   
3        21.3             2.94     YES                    9.268      11.2672   
4        21.3             5.33      NO                    8.824      11.2896   
..        ...              ...     ...                      ...          ...   
501      19.0             9.67      NO                    9.348      12.1792   
502      19.0             9.08     YES                    6.612      13.1648   
503      19.0             5.64      NO                    5.478      12.1912   
504      19.0             6.48     YES                    7.940      15.1760   
505      19.0             7.88     YES                   10.280      10.1520   

          Waterbody  Rainfall Bus_Terminal     Parks  Sold  
0             River        23          YES  0.049347     0  
1              Lake        42          YES  0.046146     1  
2              None        38          YES  0.045764     0  
3              Lake        45          YES  0.047151     0  
4              Lake        55          YES  0.039474     0  
..              ...       ...          ...       ...   ...  
501  Lake and River        27          YES  0.056006     1  
502  Lake and River        20          YES  0.059903     1  
503            None        31          YES  0.057572     1  
504            None        47          YES  0.060694     1  
505            None        45          YES  0.060336     1  

[506 rows x 16 columns]

In [10]:
df_houseprice.columns

Index(['Price', 'Residential_Area', 'Air_Qulity', 'Room_Numbers', 'Age',
       'Distance', 'Teachers', 'Poor_Population', 'Airport',
       'Number_Of_Hospital_Beds', 'n_hot_rooms', 'Waterbody', 'Rainfall',
       'Bus_Terminal', 'Parks', 'Sold'],
      dtype='object')

### b.null values handling

Laws that we need to fallow while processing the null value 
 - when we have less than 15% data is missing -> remove the observation 
 - between 20% to 70% data is missing -> impure the null value
 - greater than 75% data is missing -> remove the column 

In [11]:
# find the null values 
df_houseprice.isnull().sum()

Price                      0
Residential_Area           0
Air_Qulity                 0
Room_Numbers               0
Age                        0
Distance                   0
Teachers                   0
Poor_Population            0
Airport                    0
Number_Of_Hospital_Beds    8
n_hot_rooms                0
Waterbody                  0
Rainfall                   0
Bus_Terminal               0
Parks                      0
Sold                       0
dtype: int64

In [12]:
#persentage the missing values
round((df_houseprice.isnull().sum())/len(df_houseprice)*100,3)

Price                      0.000
Residential_Area           0.000
Air_Qulity                 0.000
Room_Numbers               0.000
Age                        0.000
Distance                   0.000
Teachers                   0.000
Poor_Population            0.000
Airport                    0.000
Number_Of_Hospital_Beds    1.581
n_hot_rooms                0.000
Waterbody                  0.000
Rainfall                   0.000
Bus_Terminal               0.000
Parks                      0.000
Sold                       0.000
dtype: float64

In [13]:
df_houseprice.Waterbody.mode()

# this is a categorical a colums so we can replace the null with mode
df_houseprice.Waterbody.fillna('river')


0               River
1                Lake
2                None
3                Lake
4                Lake
            ...      
501    Lake and River
502    Lake and River
503              None
504              None
505              None
Name: Waterbody, Length: 506, dtype: object

In [14]:
df_houseprice.Waterbody.mode()

0    River
Name: Waterbody, dtype: object

In [15]:
# drop a null value
df_houseprice.dropna(inplace =True)


## prepare for machine learning we know that machine learning algorithm so we need to provide the data to the algorithm in specific format 
# encoding 
# divide the data  based on their data types 


In [16]:
df_cat = df_houseprice.select_dtypes(include = ['object'])

df_cat.columns

Index(['Airport', 'Waterbody', 'Bus_Terminal'], dtype='object')

In [17]:
# use dummy variable from pandas to create dummy variable
# use drop first to create a  (n-1) dummy variable
dummy_var = pd.get_dummies(data = df_cat,drop_first = True)
dummy_var = dummy_var.reset_index()

In [18]:
# use dummy variable from pandas to create dummy variable
# use drop first to create a  (n-1) dummy variable
dummy_var2 = pd.get_dummies(data = df_cat)

In [19]:
dummy_var.head(2)

index  Airport_YES  Waterbody_Lake and River  Waterbody_None  \
0      0            1                         0               0   
1      1            0                         0               0   

   Waterbody_River  
0                1  
1                0

In [20]:
dummy_var2.head(5)

Airport_NO  Airport_YES  Waterbody_Lake  Waterbody_Lake and River  \
0           0            1               0                         0   
1           1            0               1                         0   
2           1            0               0                         0   
3           0            1               1                         0   
4           1            0               1                         0   

   Waterbody_None  Waterbody_River  Bus_Terminal_YES  
0               0                1                 1  
1               0                0                 1  
2               1                0                 1  
3               0                0                 1  
4               0                0                 1

In [21]:
df_num = df_houseprice.select_dtypes(include = [np.number])
df_num.head()

Price  Residential_Area  Air_Qulity  Room_Numbers   Age  Distance  \
0   24.0             32.31       0.538         6.575  65.2      4.35   
1   21.6             37.07       0.469         6.421  78.9      4.99   
2   34.7             37.07       0.469         7.185  61.1      5.03   
3   33.4             32.18       0.458         6.998  45.8      6.21   
4   36.2             32.18       0.458         7.147  54.2      6.16   

   Teachers  Poor_Population  Number_Of_Hospital_Beds  n_hot_rooms  Rainfall  \
0      24.7             4.98                    5.480      11.1920        23   
1      22.2             9.14                    7.332      12.1728        42   
2      22.2             4.03                    7.394     101.1200        38   
3      21.3             2.94                    9.268      11.2672        45   
4      21.3             5.33                    8.824      11.2896        55   

      Parks  Sold  
0  0.049347     0  
1  0.046146     1  
2  0.045764     0  
3  0.047151     0  
4  0.039474     0

In [22]:
from sklearn.preprocessing import StandardScaler

In [23]:
df_target = df_houseprice['Price']
df_num = df_num.drop('Price', axis = 1)
df_num.head()

Residential_Area  Air_Qulity  Room_Numbers   Age  Distance  Teachers  \
0             32.31       0.538         6.575  65.2      4.35      24.7   
1             37.07       0.469         6.421  78.9      4.99      22.2   
2             37.07       0.469         7.185  61.1      5.03      22.2   
3             32.18       0.458         6.998  45.8      6.21      21.3   
4             32.18       0.458         7.147  54.2      6.16      21.3   

   Poor_Population  Number_Of_Hospital_Beds  n_hot_rooms  Rainfall     Parks  \
0             4.98                    5.480      11.1920        23  0.049347   
1             9.14                    7.332      12.1728        42  0.046146   
2             4.03                    7.394     101.1200        38  0.045764   
3             2.94                    9.268      11.2672        45  0.047151   
4             5.33                    8.824      11.2896        55  0.039474   

   Sold  
0     0  
1     1  
2     0  
3     0  
4     0

In [24]:
# initiatate the standard scaler
ss = StandardScaler()

In [25]:
# transform the data 
Scale = ss.fit_transform(df_num)
Scale

array([[-1.28244435, -0.13709359,  0.40897478, ..., -1.30206786,
        -0.47313205, -0.91152748],
       [-0.59005882, -0.73319492,  0.18997385, ...,  0.21519397,
        -0.77522463,  1.09705963],
       [-0.59005882, -0.73319492,  1.276446  , ..., -0.10422957,
        -0.81123666, -0.91152748],
       ...,
       [ 0.11687263,  0.16527665,  0.97923045, ..., -0.66322077,
         0.30293265,  1.09705963],
       [ 0.11687263,  0.16527665,  0.72041117, ...,  0.6144734 ,
         0.59750894,  1.09705963],
       [ 0.11687263,  0.16527665, -0.36606097, ...,  0.45476163,
         0.56369225,  1.09705963]])

In [26]:
df_Scale = pd.DataFrame(Scale, columns = df_num.columns)
df_Scale = df_Scale.reset_index()
df_Scale

index  Residential_Area  Air_Qulity  Room_Numbers       Age  Distance  \
0        0         -1.282444   -0.137094      0.408975 -0.115644  0.172845   
1        1         -0.590059   -0.733195      0.189974  0.369741  0.476100   
2        2         -0.590059   -0.733195      1.276446 -0.260905  0.495053   
3        3         -1.301354   -0.828226      1.010516 -0.802977  1.054179   
4        4         -1.301354   -0.828226      1.222407 -0.505369  1.030487   
..     ...               ...         ...           ...       ...       ...   
493    493          0.116873    0.165277      0.434572  0.022531 -0.637413   
494    494          0.116873    0.165277     -0.238073  0.291796 -0.732180   
495    495          0.116873    0.165277      0.979230  0.798438 -0.779564   
496    496          0.116873    0.165277      0.720411  0.738208 -0.684797   
497    497          0.116873    0.165277     -0.366061  0.437057 -0.599506   

     Teachers  Poor_Population  Number_Of_Hospital_Beds  n_hot_rooms  \
0    1.462167        -1.066368                -1.640299    -0.352111   
1    0.304963        -0.484136                -0.384875    -0.166019   
2    0.304963        -1.199330                -0.342847    16.710340   
3   -0.111631        -1.351886                 0.927490    -0.337843   
4   -0.111631        -1.017382                 0.626514    -0.333593   
..        ...              ...                      ...          ...   
493 -1.176258        -0.409958                 0.981720    -0.164805   
494 -1.176258        -0.492534                -0.872945     0.022197   
495 -1.176258        -0.973995                -1.641654    -0.162528   
496 -1.176258        -0.856429                 0.027273     0.403791   
497 -1.176258        -0.660486                 1.613499    -0.549435   

     Rainfall     Parks      Sold  
0   -1.302068 -0.473132 -0.911527  
1    0.215194 -0.775225  1.097060  
2   -0.104230 -0.811237 -0.911527  
3    0.454762 -0.680402 -0.911527  
4    1.253320 -1.404723 -0.911527  
..        ...       ...       ...  
493 -0.982644  0.155116  1.097060  
494 -1.541636  0.522886  1.097060  
495 -0.663221  0.302933  1.097060  
496  0.614473  0.597509  1.097060  
497  0.454762  0.563692  1.097060  

[498 rows x 13 columns]

In [27]:
#make a dataframe indipendent variable 
df_independant = pd.concat([df_Scale,dummy_var],axis = 1)
df_independant

index  Residential_Area  Air_Qulity  Room_Numbers       Age  Distance  \
0        0         -1.282444   -0.137094      0.408975 -0.115644  0.172845   
1        1         -0.590059   -0.733195      0.189974  0.369741  0.476100   
2        2         -0.590059   -0.733195      1.276446 -0.260905  0.495053   
3        3         -1.301354   -0.828226      1.010516 -0.802977  1.054179   
4        4         -1.301354   -0.828226      1.222407 -0.505369  1.030487   
..     ...               ...         ...           ...       ...       ...   
493    493          0.116873    0.165277      0.434572  0.022531 -0.637413   
494    494          0.116873    0.165277     -0.238073  0.291796 -0.732180   
495    495          0.116873    0.165277      0.979230  0.798438 -0.779564   
496    496          0.116873    0.165277      0.720411  0.738208 -0.684797   
497    497          0.116873    0.165277     -0.366061  0.437057 -0.599506   

     Teachers  Poor_Population  Number_Of_Hospital_Beds  n_hot_rooms  \
0    1.462167        -1.066368                -1.640299    -0.352111   
1    0.304963        -0.484136                -0.384875    -0.166019   
2    0.304963        -1.199330                -0.342847    16.710340   
3   -0.111631        -1.351886                 0.927490    -0.337843   
4   -0.111631        -1.017382                 0.626514    -0.333593   
..        ...              ...                      ...          ...   
493 -1.176258        -0.409958                 0.981720    -0.164805   
494 -1.176258        -0.492534                -0.872945     0.022197   
495 -1.176258        -0.973995                -1.641654    -0.162528   
496 -1.176258        -0.856429                 0.027273     0.403791   
497 -1.176258        -0.660486                 1.613499    -0.549435   

     Rainfall     Parks      Sold  index  Airport_YES  \
0   -1.302068 -0.473132 -0.911527      0            1   
1    0.215194 -0.775225  1.097060      1            0   
2   -0.104230 -0.811237 -0.911527      2            0   
3    0.454762 -0.680402 -0.911527      3            1   
4    1.253320 -1.404723 -0.911527      4            0   
..        ...       ...       ...    ...          ...   
493 -0.982644  0.155116  1.097060    501            0   
494 -1.541636  0.522886  1.097060    502            1   
495 -0.663221  0.302933  1.097060    503            0   
496  0.614473  0.597509  1.097060    504            1   
497  0.454762  0.563692  1.097060    505            1   

     Waterbody_Lake and River  Waterbody_None  Waterbody_River  
0                           0               0                1  
1                           0               0                0  
2                           0               1                0  
3                           0               0                0  
4                           0               0                0  
..                        ...             ...              ...  
493                         1               0                0  
494                         1               0                0  
495                         0               1                0  
496                         0               1                0  
497                         0               1                0  

[498 rows x 18 columns]

In [28]:
#split the data for traingand testing a data
import statsmodels
import statsmodels.api as sm

In [29]:
df_independant = sm.add_constant(df_independant)

In [30]:
from sklearn.model_selection import train_test_split

In [31]:
X_train, X_test, y_train, y_test = train_test_split(df_independant, 
                                                    df_target, 
                                                    random_state = 10, 
                                                    test_size = 0.2)

In [32]:
print('X_train',X_train.shape)
print('y_train',y_train.shape)
print('X_test',X_test.shape)
print('y_test',y_test.shape)

X_train (398, 19)
y_train (398,)
X_test (100, 19)
y_test (100,)


In [33]:
X_train.columns

Index(['const', 'index', 'Residential_Area', 'Air_Qulity', 'Room_Numbers',
       'Age', 'Distance', 'Teachers', 'Poor_Population',
       'Number_Of_Hospital_Beds', 'n_hot_rooms', 'Rainfall', 'Parks', 'Sold',
       'index', 'Airport_YES', 'Waterbody_Lake and River', 'Waterbody_None',
       'Waterbody_River'],
      dtype='object')

# Linear regression

In [34]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

In [39]:
# create a Linear Regration model
model = LinearRegression()

# fit model to train data
model.fit(X_train, y_train)

# precticle on the testing data
predictions = model.predict(X_test)

mse = mean_squared_error(y_test,predictions)
print('Mean Squred Error is :',mse)

Mean Squred Error is : 15.596466798068043


In [45]:
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import root_mean_squared_error

In [46]:
mae = mean_absolute_error(y_test,predictions)
mae

3.1475596135117274

In [47]:
rmse = root_mean_squared_error(y_test,predictions)
rmse

3.949236229711771

# Ridge Model

In [50]:
from sklearn.linear_model import Ridge

In [52]:
df_independant['Sold']

0     -0.911527
1      1.097060
2     -0.911527
3     -0.911527
4     -0.911527
         ...   
493    1.097060
494    1.097060
495    1.097060
496    1.097060
497    1.097060
Name: Sold, Length: 498, dtype: float64